### PR TITLE
output to terminal when we are just waiting for cache uploads

### PR DESCRIPTION
**Pull request by gsoltis**
As a stopgap before we move cache uploads to the daemon, I added some terminal output if uploading is blocking process completion. 

While obviously not ideal that users are just waiting for cache population, I think it's at least better that they know that that is what is happening, rather than the process appearing to hang.

### CODE REVIEW

**Comment by gsoltis** [comment]
We could possibly improve the UX here with something like a [progress bar or spinner](https://github.com/schollz/progressbar)

**Comment by gsoltis** [comment]
Added a spinner. Here's what it looks like (only triggers after 1.5s of waiting)
spinner
.

File: cli/internal/spinner/spinner.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		return getWriterAndColor(terminal.Ui, useColor)
+	case *cli.MockUi:
+		return terminal.OutputWriter, false
+	default:
+		panic(fmt.Sprintf("unknown Ui: %v", terminal))
+	}
+}
+
+// WaitFor runs fn, and prints msg to the terminal if it takes longer
+// than initialDay to do complete. Depending on the terminal configuration, it may
```
Comment (tknickman) [inline cli/internal/spinner/spinner.go:35]:
```suggestion
// than initialDelay to complete. Depending on the terminal configuration, it may
```

**Review by tknickman — APPROVED** [review]
Looks good - one comment for a ..._comment_ 😄 

I like the approach - hopefully we won't need this anymore once we can move it to the daemon - but this is definitely a better ux for now.

File: cli/internal/spinner/spinner.go (hunk 1 above)
Comment (tknickman) [reply cli/internal/spinner/spinner.go:35]: Kodiak beat me here - addressed in https://github.com/vercel/turborepo/pull/1697

**Pull request merged**